cmake/modules/AddSwift.cmake

```diff
@@ -384,8 +384,10 @@ function(_add_host_variant_link_flags target)
     if(CMAKE_SYSTEM_NAME MATCHES Darwin)
       # See rdar://48283130: This gives 6MB+ size reductions for swift and
       # SourceKitService, and much larger size reductions for sil-opt etc.
-      target_link_options(${target} PRIVATE
-        "SHELL:-Xlinker -dead_strip")
+      if (NOT SWIFT_DISABLE_DEAD_STRIPPING)
+        target_link_options(${target} PRIVATE
+          "SHELL:-Xlinker -dead_strip")
+      endif()
```

Comment:
You could simplify the surrounding code here using the [`if(<cond1> AND <cond2>)` form](https://cmake.org/cmake/help/latest/command/if.html#condition-syntax).